src/Utility/RetryScenarioTester.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace SilverStripe\BehatExtension\Utility;
+
+use Behat\Behat\Tester\ScenarioTester;
+use Behat\Gherkin\Node\FeatureNode;
+use Behat\Gherkin\Node\ScenarioInterface;
+use Behat\Testwork\Environment\Environment;
+use Behat\Testwork\Tester\Result\TestResult;
+
+/**
+ * This tester will retry a scenario if it fails within the same test run.
+ * Note that this is different from the following retry solutions:
+ * - RetryableCallHandler using the @retry tag, which gives assertion an additional 3 seconds
+ * - RerunController, which uses the --rerun option to run only previously failed scenarios in a subsequent test run
+ */
+class RetryScenarioTester implements ScenarioTester
+{
+    private ScenarioTester $decoratedTester;
+
+    public function __construct(ScenarioTester $decoratedTester)
+    {
+        $this->decoratedTester = $decoratedTester;
+    }
+
+    public function setUp(Environment $env, FeatureNode $feature, ScenarioInterface $scenario, $skip)
+    {
+        return $this->decoratedTester->setUp($env, $feature, $scenario, $skip);
+    }
+
+    public function test(Environment $env, FeatureNode $feature, ScenarioInterface $scenario, $skip)
+    {
+        $result = $this->decoratedTester->test($env, $feature, $scenario, $skip);
+        if (!$skip && !$result->isPassed()) {
+            // Run the scenario again
+            file_put_contents('php://stdout', 'Retrying scenario: ' . $scenario->getTitle() . PHP_EOL);
+            $result = $this->decoratedTester->test($env, $feature, $scenario, $skip);
```

Comment:
This isn't enough for some scenarios.

I edited the `gridfield-import.feature` feature in `silverstripe/admin`  by making this change:
```diff
- Then I should see 1 ".col-Name" elements
+ Then I should see 1 ".col-Nameaaaa" elements
```

With this PR I get the following output:

![image](https://github.com/silverstripe/silverstripe-behat-extension/assets/36352093/aa08ba66-aac8-47ab-bc07-d575f235013e)

<details><summary>Click to see text copy of the above screenshot</summary>

```text
Bootstrapping
Creating test session environment
[WARNING]: Schema default is empty. Skipping.
[INFO]: Built schema in 2ms.
Temp Database: ss_tmpdb_1712805872_1944170

Feature: Import in GridField
  As a site owner
  I want confidence that only users with permission can import records
  So that I can sleep at night

[WARNING]: Schema default is empty. Skipping.
[INFO]: Built schema in 1ms.
  Background:                                              # vendor/silverstripe/admin/tests/behat/features/gridfield-import.feature:6
    Given the "Company" "Walmart" with "Category"="Retail" # SilverStripe\BehatExtension\Context\FixtureContext::stepCreateRecordWithData()
    And I am logged in with "ADMIN" permissions            # SilverStripe\BehatExtension\Context\LoginContext::iAmLoggedInWithPermissions()

  @gsat
  Scenario: I can create new records via the import form                       # vendor/silverstripe/admin/tests/behat/features/gridfield-import.feature:11
    When I go to "/admin/test"                                                 # SilverStripe\Framework\Tests\Behaviour\FeatureContext::visit()
Saving screenshot into /var/www/html/.ddev-extra/artifacts/gridfield-import.feature_13.png
Saving HTML into /var/www/html/.ddev-extra/artifacts/zz_gridfield-import.feature_13.html
    Then I should see 1 ".col-Nameaaaa" elements                               # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertNumElements()
      0 elements matching css ".col-Nameaaaa" found on the page, but should be 1. (Behat\Mink\Exception\ExpectationException)
    And I press the "Import CSV" button                                        # SilverStripe\BehatExtension\Context\BasicContext::stepIPressTheButton()
    Then I should see the "#Form_ImportForm" element                           # SilverStripe\BehatExtension\Context\BasicContext::iShouldSeeTheElement()
    When I attach the file "import-company-create.csv" to the "_CsvFile" field # SilverStripe\BehatExtension\Context\BasicContext::iAttachTheFileToTheField()
    And I press the "Import from CSV" button                                   # SilverStripe\BehatExtension\Context\BasicContext::stepIPressTheButton()
    Then I should see "Imported 1 record" in the "#Form_ImportForm" element    # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertElementContainsText()
    And I should see 2 ".col-Name" elements                                    # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertNumElements()
Retrying scenario: I can create new records via the import form
Saving screenshot into /var/www/html/.ddev-extra/artifacts/gridfield-import.feature_8.png
Saving HTML into /var/www/html/.ddev-extra/artifacts/zz_gridfield-import.feature_8.html
    And I am logged in with "ADMIN" permissions                                # SilverStripe\BehatExtension\Context\LoginContext::iAmLoggedInWithPermissions()
      Could not find login email field
      Failed asserting that null is not null.
    When I go to "/admin/test"                                                 # SilverStripe\Framework\Tests\Behaviour\FeatureContext::visit()
    Then I should see 1 ".col-Nameaaaa" elements                               # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertNumElements()
    And I press the "Import CSV" button                                        # SilverStripe\BehatExtension\Context\BasicContext::stepIPressTheButton()
    Then I should see the "#Form_ImportForm" element                           # SilverStripe\BehatExtension\Context\BasicContext::iShouldSeeTheElement()
    When I attach the file "import-company-create.csv" to the "_CsvFile" field # SilverStripe\BehatExtension\Context\BasicContext::iAttachTheFileToTheField()
    And I press the "Import from CSV" button                                   # SilverStripe\BehatExtension\Context\BasicContext::stepIPressTheButton()
    Then I should see "Imported 1 record" in the "#Form_ImportForm" element    # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertElementContainsText()
    And I should see 2 ".col-Name" elements                                    # SilverStripe\Framework\Tests\Behaviour\FeatureContext::assertNumElements()

--- Failed scenarios:

    vendor/silverstripe/admin/tests/behat/features/gridfield-import.feature:11

1 scenario (1 failed)
20 steps (4 passed, 2 failed, 14 skipped)
0m7.03s (16.65Mb)
Killing test session environment... done!
Failed to run behat admin: exit status 1
```
</details>

Looks like we need to reset after the first scenario fails, rerun the background, and go from there.